runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useCustomValidation.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback } from 'react';
+
+import { useDispatch, useSelector } from '#src-app/store';
+import { ModelerErrorType, processActions } from '#src-app/store/slices/Process';
+import getElementLabel from '#src-app/utils/getElementLabel';
+
+type CustomValidator = () => boolean;
+
+const useCustomValidation = () => {
+    const { selectedElement } = useSelector(state => state.process.modeler);
+    const dispatch = useDispatch();
+
+    const validator = useCallback((customValidator: CustomValidator) => {
+        const isValid = customValidator();
+        if (isValid) {
+            dispatch(processActions.removeCustomValidationError(selectedElement.id));
+        } else {
+            dispatch(processActions.setCustomValidationError({
+                elementId: selectedElement.id,
+                elementName: getElementLabel(selectedElement),
+                type: ModelerErrorType.FORM_ERROR,
+            }));
+        }
+        return isValid;
+    }, []);
+
+    return { validator };
+};
```

Comment:
here you depend on `selectedElement` imo you should include it in `useCallback` deps